SparseGrids/tsgCudaMacros.hpp

```diff
@@ -58,47 +54,37 @@ namespace TasGrid{
 namespace TasCUDA{
     // general GPU/CPU I/O, new/delete vectors, send/recv data
     template <typename T>
-    inline T* cudaNew(int num_entries, std::ostream *os = 0){
+    inline T* cudaNew(size_t num_entries, std::ostream *os){
```

Comment:
How about you pass it by reference then?

Reply:
You mean the ostream? The specifies the ostream in a TasmanianSparseGrid class and then it gets propagated through several functions. The use can pass a NULL pointer to disable logging altogether, the code checks "if (os == 0)" in the cudaCheckError().